Apps/WebClient/src/ClientApp/src/components/report/covid19.vue

```diff
@@ -0,0 +1,248 @@
+                            results.resultError
+                        )
+                    );
+                }
+            })
+            .catch((err) => {
+                this.logger.error(err);
+                this.addError(
+                    ErrorTranslator.toBannerError(
+                        "Fetch Laboratory Results Error",
+                        err
+                    )
+                );
+            })
+            .finally(() => {
+                this.isLoading = false;
+            });
+    }
+
+    private sortEntries() {
+        this.records.sort((a, b) =>
+            a.messageDateTime > b.messageDateTime
+                ? -1
+                : a.messageDateTime < b.messageDateTime
+                ? 1
+                : 0
+        );
+    }
+
+    private get currentDate() {
+        return moment(new Date()).format("ll");
+    }
+
+    private get isEmpty() {
+        return this.records.length == 0;
+    }
+
+    private formatDate(date: Date): string {
+        return moment(date).format("yyyy-MM-DD");
+    }
+
+    private mounted() {
+        this.logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+        this.fetchLaboratoryResults();
+    }
+
+    public async generatePdf(): Promise<void> {
+        this.logger.debug("generating COVID-19 PDF...");
+        this.isLoading = true;
+
+        let opt = {
+            margin: [25, 15],
+            filename: `HealthGateway_COVID19.pdf`,
+            image: { type: "jpeg", quality: 1 },
+            html2canvas: { dpi: 192, scale: 1.1, letterRendering: true },
+            jsPDF: { unit: "pt", format: "letter", orientation: "portrait" },
+            pagebreak: { mode: ["avoid-all", "css", "legacy"] },
+        };
+        html2pdf()
+            .set(opt)
+            .from(this.report)
+            .toPdf()
+            .get("pdf")
+            .then((pdf: PDFDefinition) => {
+                // Add footer with page numbers
+                var totalPages = pdf.internal.getNumberOfPages();
+                for (let i = 1; i <= totalPages; i++) {
+                    pdf.setPage(i);
+                    pdf.setFontSize(10);
+                    pdf.setTextColor(150);
+                    pdf.text(
+                        `Page ${i} of ${totalPages}`,
+                        pdf.internal.pageSize.getWidth() / 2 - 55,
+                        pdf.internal.pageSize.getHeight() - 10
+                    );
+                }
+            })
+            .save()
+            .output("bloburl")
+            .then((pdfBlobUrl: RequestInfo) => {
+                fetch(pdfBlobUrl).then((res) => {
+                    res.blob().then(() => {
+                        this.isLoading = false;
+                    });
+                });
+            });
+    }
+}
+</script>
+
+<template>
+    <div>
+        <LoadingComponent :is-loading="isLoading"></LoadingComponent>
+        <div ref="report">
+            <section class="pdf-item">
```

Comment:
I pushed a change that makes the header conditional, look at the changes in the PR below and apply those to this report too!
https://github.com/bcgov/healthgateway/pull/1425/files

Reply:
updated